tapioca/adapters.py

```diff
@@ -109,3 +111,47 @@ def format_data_to_request(self, data):
     def response_to_native(self, response):
         if response.content.strip():
             return response.json()
+
+
+class XMLAdapterMixin(object):
+
+    def __init__(self, serializer_class=None, *args, **kwargs):
+        super(XMLAdapterMixin, self).__init__(serializer_class, *args, **kwargs)
+
+    def _input_branches_to_xml_bytestring(self, data):
+        if isinstance(data, Mapping):
+            return xmltodict.unparse(
+                data, **self._xmltodict_unparse_kwargs).encode('utf-8')
+        try:
+            return data.encode('utf-8')
+        except Exception as e:
+            raise type(e)('Format not recognized, please enter an XML as string or a dictionary'
+                          'in xmltodict spec: \n%s' % e.message)
+
+    def get_request_kwargs(self, api_params, *args, **kwargs):
+        # stores kwargs prefixed with 'xmltodict_unparse__' for use by xmltodict.unparse
+        self._xmltodict_unparse_kwargs = {k[len('xmltodict_unparse__'):]: kwargs.pop(k)
+                                          for k in kwargs.copy().keys()
+                                          if k.startswith('xmltodict_unparse__')}
+        # stores kwargs prefixed with 'xmltodict_parse__' for use by xmltodict.parse
+        self._xmltodict_parse_kwargs = {k[len('xmltodict_parse__'):]: kwargs.pop(k)
+                                        for k in kwargs.copy().keys()
+                                        if k.startswith('xmltodict_parse__')}
+
+        arguments = super(XMLAdapterMixin, self).get_request_kwargs(
+            api_params, *args, **kwargs)
+
+        if 'headers' not in arguments:
+            arguments['headers'] = {}
+        arguments['headers']['Content-Type'] = 'application/xml'
+        return arguments
+
+    def format_data_to_request(self, data):
+        if data:
+            return self._input_branches_to_xml_bytestring(data)
+
+    def response_to_native(self, response):
+        if response.content.strip():
+            if 'xml' in response.headers['content-type']:
```

Comment:
@filipeximenes the cause for adding this check is a cryptic `ExpatError` that is thrown whenever the server doesn't return valid XML.

The Google Sheets endpoint I'm testing with returns a string when it encounters an error condition (e.g. 400, 401 error), and Tapioca's `ClientError` is interrupted and replaced with the `ExpatError`:

```
ExpatError: syntax error: line 1, column 0
```

For non-error conditions, I thought it best to return the text of the response so that an end-user could inspect it, instead of returning nothing. We could use a try / except statement around `xmltodict.parse()` and handle any exception that xmltodict throws, rather than indirectly handling it through checking the response headers, but we need something here.